File: go/easy/3875_construct_uniform_parity_array_i.go
func uniformArray(nums1 []int) bool {
    check := func(nums []int, predicate func(int) bool) bool {
        for i, u := range nums {
            if predicate(u) {
                continue
            }
            p := false
            for j, v := range nums {
                if i == j {
                    continue
                }
                if predicate(u + u + v) {
                    p = true
                    break
                }
            }
            if !p {
                return false
            }
        }
        return true
    }
    even := check(nums1, func(n int) bool {
        return n % 2 == 0
    })
    odd := check(nums1, func(n int) bool {
        return n % 2 != 0
    })
    return odd || even  
}

File: rust/easy/3875_construct_uniform_parity_array_i.rs
impl Solution {
    pub fn uniform_array(nums1: Vec<i32>) -> bool {
        fn check(nums: &[i32], predicate: fn(i32) -> bool) -> bool {
            for i in 0..nums.len() {
                if predicate(nums[i]) {
                    continue
                }
                let mut possible = false;
                for j in 0..nums.len() {
                    if i == j {
                        continue;
                    }
                    if predicate(nums[i] + nums[i] - nums[j]) {
                        possible = true;
                        break;
                    }
                }
                if !possible {
                    return false;
                }
            }
            true
        }
        let even = check(&nums1, |n| n % 2 == 0);
        let odd = check(&nums1, |n| n % 2 != 0);
        odd || even
    }
}
